### DISCUSSION TITLE
Makna c('value1' = 'value2')

### DISCUSSION THREAD

**Discussion by myibra** [teamalgoritma/community#96] [Q&A] [answered]
Selamat siang semuanya, 

saya ingin bertanya terkait code dari https://shiny.rstudio.com/gallery/navbar-example.html.

berikut adalah code pada bagian `ui`:
` radioButtons("plotType", "Plot type",
          c("Scatter"="p", "Line"="l")
        )`

sedangkan dibawah adalah code pada bagian `server`:
`  output$plot <- renderPlot({
    plot(cars, type=input$plotType)
  })`

saya ingin menekankan ke bagian `c("Scatter"="p", "Line"="l")` menurut pemahaman saya di code tersebut akan menampilkan 'Scatter' dan 'Line' pada fungsi `radioButtons()` *tetapi* akan mengembalikan nilai 'p' dan 'l' di fungsi `renderPlot()`. Jika saya ingin menspesifik kan penggunaan value dari `c("Scatter"="p", "Line"="l")`? sebagai contoh jika saya ingin menggunakan value 'Scatter' dan 'Line' sebagai judul plot.

Terimakasih

**Comment by VictorNugraha** [comment, marked as answer]
Halo, @myibra!

Disini saya setuju dengan pernyataan Bapak mengenai, `c("Scatter"="p", "Line"="l")` akan menampilkan 'Scatter' dan 'Line' pada fungsi `radioButtons()` dan akan mengembalikan nilai **_p_** dan **_l_** di fungsi `renderPlot()`.

Untuk pertanyaan lanjutan dari Bapak, bagaimana memanfaatkan value `c("Scatter"="p", "Line"="l")`, agar bisa menjadi judul dari plot yang dihasilkan? Semoga saya tidak salah dalam menafsirkan maksud dari pertanyaan yang diberikan, Bapak bisa menambahkan code yang saya lampirkan di bawah ini setelah fungsi **_plot()_**.

```
    title(paste("Plot Typle: ", 
                ifelse(test = input$plotType == "p",  # Parameter test digunakan untuk membuat sebuah kondisi if. 
                                                        Pembuatan kondisi if akan menggunakan input$plotType yang dibuat di radioButton() karena dari input tersebutlah plot yang dihasilkan akan berubah.
                       yes = print("Scatter"),  # Parameter yes digunakan sebagai hasil yang diharapkan ketika kondisi if-nya terpenuhi
                       no = print("Line")) # Parameter no digunakan sebagai hasil yang diharapkan ketika kondisi if-nya tidak terpenuhi
           ))
```

Dimana,

- Fungsi _**title()**_: Fungsi ini digunakan untuk menambahkan title, jika kita menggunakan fungsi _**plot()**_ untuk membuat visualisasinya. 
- Fungsi _**paste()**_: Fungsi ini digunakan untuk menggabungkan beberapa nilai/vector yang nantinya akan mengasilkan judul dari plot.
- Fungsi _**ifelse()**_: Fungsi ini digunakan untuk membuat kondsi if-else. Disini kita membutuhkan sebuah kondisi if-else karena nilai yang akan dikembalikan ke visualisasinya adalah nilai **_p_** dan **_l_**.

Demikian informasi yang dapat kami sampaikan, semoga menjawab pertanyaan yang diberikan. Selamat mencoba :D

**Reply by myibra** [reply]
jadi hasil dari code `print("Scatter")` akan ter print setelah ':' di `"Plot Typle: "` ya mas?

Bagaimana ketika kondisinya tidak 2 mas? apakah fungsi _*ifelse()*_ masih_ bisa mengakomodir jika dilakukan bertingkat?

**Reply by VictorNugraha** [reply]
Betul sekali Pak, hasil dari code `print("Scatter")` akan muncuh setelah tulisan _**Plot Type:**_.

Jika kondisinya lebih dari 2, Bapak bisa menambakan fungsi _**ifelse()**_ lagi pada parameter _**no =**_ di fungsi _**ifelse()**_ yang pertama.

Disini ada opsi fungsi lainnya, jika dirasa terlalu banyak fungsi _**ifelse()**_. Bapak bisa menggunakan funsi **_case_when()_** sebagai alternaifnya. Berikut contoh codenya:

```
case_when(condition_1 ~ output_value1,
          condition_2 ~ output_value2,
          condition_3 ~ output_value3)
```

Demikian informasi yang dapat kami sampaikan, semoga menjawab pertanyaan yang diberikan. Selamat mencoba :)
